File: Domain/Book.cs
namespace Domain
{
    public class Book
    {
        public Guid BookId { get; set; }
        public string ISBN { get; set; }
        public string Title { get; set; }
        public string Image { get; set; }
        public int NoOfPages { get; set; }
        public int YearOfPublishing { get; set; }
        public DateOnly PurshaseDate { get; set; }
        public decimal Price { get; set; }
        public decimal Rate { get; set; }
        public string Description { get; set; }
        public bool IsRead { get; set; }
        public Publisher Publisher { get; set; }
        public Guid PublisherId { get; set; }
        public Format Format { get; set; }
        public Guid FormatId { get; set; }
        public string AppUserId { get; set; }
        public AppUser AppUser { get; set; }
        public ICollection<BookAuthors> BookAuthors { get; set; } = new List<BookAuthors>();
        public ICollection<BookGenres> Genres { get; set; } = new List<BookGenres>();
        public ICollection<BookSeries> Series { get; set; } = new List<BookSeries>();
    }
}

File: client-app/src/app/models/book.ts
import { Author } from "./author";
import { Format } from "./format";
import { Profile } from "./profile";
import { Publisher } from "./publisher";

export interface Book {
    bookId: string;
    title: string;
    isbn: string;
    image: string;
    purchaseDate: Date | null;
    noOfPages: number;
    yearOfPublishing: number;
    price: number;
    rate: number | 0;
    description: string;
    isRead: boolean;
    profile: Profile;
    publisher: Publisher;
    authors: Author[];
    format: Format;
}

export class Book implements Book {
    constructor(init?: BookFormValues) {
        Object.assign(this, init);
    }
}

export class BookFormValues {
    bookId?: string = undefined;
    title: string = "";
    isbn: string = "";
    image: string = "";
    purchaseDate: Date | null = null;
    noOfPages: number = 0;
    yearOfPublishing: number = 0;
    price: number = 0;
    rate: number | 0 = 0;
    description: string = "";
    isRead: boolean = false;

    constructor(book?: BookFormValues) {
        if(book) {
            this.bookId = book.bookId;
            this.title = book.title;
            this.isbn = book.isbn;
            this.image = book.image;
            this.purchaseDate = book.purchaseDate;
            this.noOfPages = book.noOfPages;
            this.yearOfPublishing = book.yearOfPublishing;;
            this.price = book.price;
            this.rate = book.rate;
            this.description = book.description;
            this.isRead = book.isRead;
        }
    }
}
